core/src/main/java/hudson/model/AbstractProject.java

```diff
@@ -1922,9 +1922,8 @@ public DirectoryBrowserSupport doWs( StaplerRequest req, StaplerResponse rsp ) t
      */
     public HttpResponse doDoWipeOutWorkspace() throws IOException, ServletException, InterruptedException {
         checkPermission(Functions.isWipeOutPermissionEnabled() ? WIPEOUT : BUILD);
-        R b = getSomeBuildWithWorkspace();
-        FilePath ws = b!=null ? b.getWorkspace() : null;
-        if (ws!=null && getScm().processWorkspaceBeforeDeletion(this, ws, b.getBuiltOn())) {
```

Comment:
To clarify, only one overload (the one for `AbstractProject` rather than `Job`) is deprecated after changes to the model hierarchy. This doesn't mean `processWorkspaceBeforeDeletion` shouldn't be called at all. You could just cast `this` to `Job` to get rid of the deprecation warning.